2004-12-13  Jeff Johnston  <jjohnstn@redhat.com>

	* dwarf2read.c (read_type_die): Ensure that structures, unions,
	enumerations, and base types create symbols.

Index: gdb-6.8/gdb/dwarf2read.c
===================================================================
--- gdb-6.8.orig/gdb/dwarf2read.c	2008-03-10 15:18:10.000000000 +0100
+++ gdb-6.8/gdb/dwarf2read.c	2008-07-14 10:26:01.000000000 +0200
@@ -7711,9 +7711,11 @@ read_type_die (struct die_info *die, str
     case DW_TAG_structure_type:
     case DW_TAG_union_type:
       read_structure_type (die, cu);
+      process_structure_scope (die, cu);
       break;
     case DW_TAG_enumeration_type:
       read_enumeration_type (die, cu);
+      process_enumeration_scope (die, cu);
       break;
     case DW_TAG_subprogram:
     case DW_TAG_subroutine_type:
@@ -7751,6 +7753,7 @@ read_type_die (struct die_info *die, str
       break;
     case DW_TAG_base_type:
       read_base_type (die, cu);
+      new_symbol (die, die->type, cu);
       break;
     case DW_TAG_unspecified_type:
       read_unspecified_type (die, cu);
